druid/src/widget/images.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An Image widget.
+//! Please consider using Image and the Image wideget as it scales much better.
+
+use std::error::Error;
+use std::marker::PhantomData;
+use std::str::FromStr;
+
+use crate::{
+    Affine, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx,
+    PaintCtx, Point, Rect, RenderContext, Size, UpdateCtx, Widget,
+};
+
+use crate::piet::{ImageFormat, InterpolationMode};
+
+use image;
+
+#[derive(PartialEq)]
+pub enum FillStrat {
+    Contain,
+    Cover,
+    Fill,
+    FitHeight,
+    FitWidth,
+    None,
+    ScaleDown,
+}
+
+pub enum FloatVert {
```

Comment:
these two don't seem to be really used?

In any case I would prefer to not be adding new types here, if possible. It feels to me like this could be the responsibility of another widget? Or alternatively maybe we could use the `UnitPoint` type?

Reply:
I was gona use these but i think they might be better implemented as widgets, i can put them back and use them if needed, if any one wants them i am happy to implement them.

They could be used so that when dead space is created due to the requested fill type. The dead space can be biased to one side rather than the current implementation of always having it split across both sides.

Reply:
If we want this behaviour I would use `UnitPoint` to determine where we align.